crates/noirc_abi/src/input_parser/toml.rs

```diff
@@ -82,67 +113,27 @@ fn toml_map_to_field(
                 InputValue::Vec(array_elements)
             }
             TomlTypes::ArrayString(arr_str) => {
-                let array_elements: Vec<_> = arr_str
-                    .into_iter()
-                    .map(|elem_str| parse_str(&elem_str).unwrap().unwrap())
-                    .collect();
+                let array_elements = try_vecmap(arr_str, |elem_str| match parse_str(&elem_str) {
+                    Ok(Some(elem_value)) => Ok(elem_value),
+                    Ok(None) => {
+                        Err(InputParserError::ParseStr("undefined array element".to_string()))
+                    }
+                    Err(err) => Err(err),
```

Comment:
Nitpicking but we can replace this match with

```rs
parse_str(&elem_str)?.ok_or_else(|| 
    InputParserError::ParseStr("undefined array element".to_string()
)
```

Reply:
Come to think of it, it'd likely be better to have `parse_str` return a proper error type for empty strings. We currently need to have special handling for them anyway (L91-95) and when string support is added we won't need to parse the string anyway.